sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_task_adherence/task_adherence.prompty

```diff
@@ -25,93 +25,102 @@ inputs:
+**Definition:** Response addresses the main task and includes some relevant content, but key elements are missing or underdeveloped. It may lack completeness, structure, or sufficient clarity to fully satisfy the task.
 
 **Example:**
-  **Query:** What is a recommended weekend itinerary in Paris?
-  **Response:** Visit the Eiffel Tower and the Louvre on Saturday, and stroll through Montmartre on Sunday.
+    **Query:** Recommend a historical fiction book.
+    **Response:** You might like **The Book Thief** by Markus Zusak.
+    **Tool Definitions:** [{ **name**: "get_book", **description**: "Retrieve a book recommendation for a specified genre.", **parameters**: { "genre": "string" } }]
 
-Explanation: This answer meets the basic requirement by naming a few key attractions and assigning them to specific days. However, it lacks additional context, such as timings, additional activities, or details to make the itinerary practical and clear.
+    **Explanation:** The assistant gives a relevant book, showing moderate task adherence. However, it does not use the tool or give supporting information.
 
+## [TaskAdherence: 4] (Mostly Adherent)
 
-**Level 4: Mostly Adherent**
+**Definition:** Response is clear, accurate, and aligns with instructions with minor issues.
 
-**Definition:**
-Response is clear, accurate, and aligns with instructions with minor issues.
+    **Example:**
+    **Query:** Recommend a historical fiction book.
+    **Response:**
+    Let me check...
+    → **[Tool Call: get_book({ "genre": "historical fiction" })]**
+    → **[Tool Result: { "title": "Wolf Hall", "author": "Hilary Mantel" }]**
+    You might enjoy **Wolf Hall** by Hilary Mantel.
 
-**Example:**
-  **Query:** What is a recommended weekend itinerary in Paris?
-  **Response:** For a weekend in Paris, start Saturday with a morning visit to the Eiffel Tower, then head to the Louvre in the early afternoon. In the evening, enjoy a leisurely walk along the Seine. On Sunday, begin with a visit to Notre-Dame Cathedral, followed by exploring the art and cafés in Montmartre. This plan offers a mix of cultural visits and relaxing experiences.
+    **Tool Definitions:** [{ **name**: "get_book", **description**: "Retrieve a book recommendation for a specified genre.", **parameters**: { "genre": "string" } }]
 
-Explanation: This response is clear, structured, and provides a concrete itinerary with specific attractions and a suggested order of activities. It is accurate and useful, though it might benefit from a few more details like exact timings or restaurant suggestions to be perfect.
+**Explanation:** The tool is correctly used and the result is relevant. The response is solid but could be improved with more context.
 
+## [TaskAdherence: 5] (Fully Adherent)
 
-**Level 5: Fully Adherent**
-
-**Definition:**
-Response is flawless, accurate, and follows instructions to the letter.
+**Definition:** Response is flawless, accurate, and follows instructions to the letter.
```

Comment:
We are not discussing anything about the `constraints` so far. How do we show adherence or inadherence to a constraint. How would the constraint be accepted? Could you add one more example at each level showing the incoming constraint in User Query, following by Response and then Task Adherence evaluation result covering adherence/ inadherence score.